components/lcd/esp_lcd_gc9503/include/esp_lcd_gc9503.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/**
+ * @file
+ * @brief ESP LCD: GC9503
+ */
+
+#pragma once
+
+#include "esp_lcd_types.h"
+#include "esp_lcd_panel_rgb.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Create LCD panel for GC9503.
+ *
+ * @note  This function first initialize GC9503 with vendor specific initialization, then calls `esp_lcd_new_rgb_panel()` to create a RGB LCD panel.
+ * @note  Vendor specific initialization can be different between manufacturers, should consult the LCD supplier for initialization sequence code.
+ *
+ * @param[in]  io_handle  LCD panel IO handle
+ * @param[in]  rgb_config Pointer to RGB panel timing configuration structure
+ * @param[out] ret_panel  Returned LCD panel handle
+ * @return
+ *      - ESP_OK: Success, otherwise returns ESP_ERR_xxx
+ */
+esp_err_t esp_lcd_new_panel_gc9503(esp_lcd_panel_io_handle_t io_handle, const esp_lcd_rgb_panel_config_t *rgb_config, esp_lcd_panel_handle_t *ret_panel);
```

Comment:
One more comment: this API is different to what usual esp_lcd driver looks like: 
``` c
esp_lcd_new_panel_xxx(const esp_lcd_panel_io_handle_t io, const esp_lcd_panel_dev_config_t *panel_dev_config, esp_lcd_panel_handle_t *ret_panel);
```
Your API depends on RGB implementation, while the esp_lcd driver should be IO agnostic. This has few drawbacks:
* Can be compiled only on ESP32-S3
* We can't use usual esp_lcd API like `esp_lcd_panel_reset()` ...

Reply:
Here are explanations for your concerns:
* The GC9503 only has RGB and MIPI (which is not supported) interface. So it should be compiled only on ESP32-S3.
* Actually, we can use the most esp_lcd APIs. Because the `ret_panel`  is created by `esp_lcd_new_rgb_panel()` which is in [RGB driver](https://github.com/espressif/esp-idf/blob/67552c31dac8cd94fb0d63192a538f4f984c5b6e/components/esp_lcd/src/esp_lcd_panel_rgb.c#L69). And the driver has implemented many esp_lcd APIs.

Reply:
Thank you for the explanation! I'll find a way how to build this for S3 only in CI

Reply:
in test_app/CMakeLists.txt you must exclude this component from build. Something like this: (not tested)
```
if(NOT "${IDF_TARGET}" STREQUAL "esp32s3")
    set(EXCLUDE_COMPONENTS "esp_lcd_gc9503")
endif()
```